Makefile

```diff
@@ -598,8 +620,12 @@ endif # GGML_CUDA_DEBUG
 ifdef GGML_CUDA_NVCC
 	NVCC = $(CCACHE) $(GGML_CUDA_NVCC)
 else
-	NVCC = $(CCACHE) nvcc
-endif #GGML_CUDA_NVCC
+	ifdef GGML_MUSA
+		NVCC = $(CCACHE) mcc
```

Comment:
Did you confirm whether the MUSA compiler actually works with ccache?

Reply:
Yes. I can confirm that.

```bash
root@8cb6823fed5c:/ws# make GGML_MUSA=1
I ccache found, compilation results will be cached. Disable with LLAMA_NO_CCACHE.
I llama.cpp build info: 
I UNAME_S:   Linux
I UNAME_P:   x86_64
I UNAME_M:   x86_64
I CFLAGS:    -Iggml/include -Iggml/src -Iinclude -Isrc -Icommon -D_XOPEN_SOURCE=600 -D_GNU_SOURCE -DNDEBUG -DGGML_USE_MUSA -DGGML_USE_LLAMAFILE -DGGML_USE_CUDA -I/usr/local/musa/include  -std=c11   -fPIC -O3 -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wshadow -Wstrict-prototypes -Wpointer-arith -Wmissing-prototypes -Werror=implicit-int -Werror=implicit-function-declaration -pthread -march=native -mtune=native -Wunreachable-code-break -Wunreachable-code-return -Wdouble-promotion 
I CXXFLAGS:  -std=c++11 -fPIC -O3 -Wall -Wextra -Wpedantic -Wcast-qual -Wno-unused-function -Wmissing-declarations -Wmissing-noreturn -pthread  -march=native -mtune=native -Wunreachable-code-break -Wunreachable-code-return -Wmissing-prototypes -Wextra-semi -Iggml/include -Iggml/src -Iinclude -Isrc -Icommon -D_XOPEN_SOURCE=600 -D_GNU_SOURCE -DNDEBUG -DGGML_USE_MUSA -DGGML_USE_LLAMAFILE -DGGML_USE_CUDA -I/usr/local/musa/include 
I NVCCFLAGS: -std=c++11 -O3 -x musa -mtgpu --cuda-gpu-arch=mp_21 -arch=native -DGGML_CUDA_DMMV_X=32 -DGGML_CUDA_MMV_Y=1 -DK_QUANTS_PER_ITERATION=2 -DGGML_CUDA_PEER_MAX_BATCH_SIZE=128 
I LDFLAGS:   -lmusa -lmublas -lmusart -lpthread -ldl -lrt -L/usr/local/musa/lib -L/usr/lib64 
I CC:        clang version 14.0.0 (git@sh-code.mthreads.com:sw/mtcc.git d4847756f2c727647d3956a6a7713ce1ef175967)
I CXX:       clang version 14.0.0 (git@sh-code.mthreads.com:sw/mtcc.git d4847756f2c727647d3956a6a7713ce1ef175967)
I NVCC:      InstalledDir: /usr/local/musa/bin

...

/usr/bin/ccache mcc -std=c++11 -O3 -x musa -mtgpu --cuda-gpu-arch=mp_21 -arch=native -DGGML_CUDA_DMMV_X=32 -DGGML_CUDA_MMV_Y=1 -DK_QUANTS_PER_ITERATION=2 -DGGML_CUDA_PEER_MAX_BATCH_SIZE=128  -Iggml/include -Iggml/src -Iinclude -Isrc -Icommon -D_XOPEN_SOURCE=600 -D_GNU_SOURCE -DNDEBUG -DGGML_USE_MUSA -DGGML_USE_LLAMAFILE -DGGML_USE_CUDA -I/usr/local/musa/include  -c ggml/src/ggml-cuda/fattn-tile-f32.cu -o ggml/src/ggml-cuda/fattn-tile-f32.o
clang-14: warning: argument unused during compilation: '-arch=native' [-Wunused-command-line-argument]
In file included from ggml/src/ggml-cuda/fattn-tile-f32.cu:1:
ggml/src/ggml-cuda/common.cuh:554:1: warning: function declared 'noreturn' should not return [-Winvalid-noreturn]
}
^
ggml/src/ggml-cuda/common.cuh:658:1: warning: non-void function does not return a value [-Wreturn-type]
}
```